app/aws/Federation.scala

```diff
@@ -42,7 +42,7 @@ object Federation {
   def duration(
       permission: Permission,
       requestedSeconds: Option[Duration] = None,
-      timezone: Option[DateTimeZone] = None
+      clock: Option[Clock] = None
```

Comment:
Using a clock here instead of a timezone so that we can pass in either a real system clock representing a changing time in a particular timezone or a test fixed clock that has stopped at a given time.